pallets/identity/src/lib.rs

```diff
@@ -1060,8 +1104,13 @@ impl<T: Config> Module<T> {
             Error::<T>::AlreadyLinked,
         );
 
-        // Replace primary key of the owner that initiated key rotation
+        // Get the current primary key
         let old_primary_key = Self::did_records(&rotation_for_did).primary_key;
+
+        // ensure that it is safe to unlink the primary key from the did.
+        Self::ensure_unlink_account_key_from_did_is_safe(&old_primary_key)?;
+
+        // Replace primary key of the owner that initiated key rotation
```

Comment:
Preexisting, but consider indenting this out using `::insert(...)` instead (since we just fetched the `record` above) to reduce one read.

Reply:
I am not sure what you mean here.  We can't move the `ensure_unlink...` check inside the `::mutate` call, since it needs to be able to throw an error.

Reply:
What I mean is basically replacing the `::mutate(...)` call here with a call to `::insert`, with something like:
```rust
let mut record = Self::did_records(&rotation_for_did);
let old_pk = record.primary_key;

Self::ensure_key_unlinkable_from_did(&old_pk)?;

Self::unlink_account_key_from_did(&old_pk, rotation_for_did);
Self::link_account_key_to_did(&sender, rotation_for_did);
record.primary_key = sender.clone();
<DidRecords<T>>::insert(&rotation_for_did, record);
```
This removes one read or at least one decode.